pan3d/dataset_builder.py

```diff
@@ -257,6 +254,48 @@ def mesh(
     # Internal methods
     # -----------------------------------------------------
 
+    def _load_dataset(self, dataset_info):
```

Comment:
maybe break that function into `_load_dataset_pangeo(...)`, `_load_dataset_esgf(...)`, `_load_dataset_xarray(...)`, `_load_dataset_url(...)`

Reply:
Those specific load_dataset method could be part of the dedicated modules (esg/pangeo/...) and just expect the builder as arg?

Reply:
I separated the default load function in [c50724f](https://github.com/Kitware/pan3d/pull/67/commits/c50724fea8b91abbe4097e5ebdddbe132d1d94b3), but the pangeo and esgf functions are already separated from `builder._load_dataset` as much as they can be. We need the builder to determine whether the module should be imported or not, so the builder class will not raise import errors if the user is missing the optional dependencies for those modules.

Reply:
what about a simple

```
try:
   import ...
  load
except:
  raise(...)
```

Reply:
Ok, I changed these to try-except blocks in [e820d80](https://github.com/Kitware/pan3d/pull/67/commits/e820d8084596b80c231f817b3938518f229a2783)